pkgs/os-specific/linux/wireguard/default.nix

```diff
@@ -35,6 +35,8 @@ let
     KERNELDIR = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";
     INSTALL_MOD_PATH = "\${out}";
 
+    NIX_CFLAGS = ["-Wno-error=cpp"];
```

Comment:
This makes it compile on older kernel, but also discard the original warnings. We can show warnings in modules (the module which sets up an wireguard interface) when older kernel version are used. However there is no guarantee that users will use the module instead of manual configuration or wg-quick.